avogadro/io/cjsonformat.cpp

```diff
@@ -422,6 +426,11 @@ bool CjsonFormat::deserialize(std::istream& file, Molecule& molecule,
         Real beta = static_cast<Real>(unitCell["beta"]) * DEG_TO_RAD;
         Real gamma = static_cast<Real>(unitCell["gamma"]) * DEG_TO_RAD;
         unitCellObject = new Core::UnitCell(a, b, c, alpha, beta, gamma);
+        if (!unitCellObject->isRegular()) {
+          appendError(
+            "cell parameters do not give linear-independent lattice vectors");
+          return false;
```

Comment:
`delete unitCellObject;` before `return false;`

Reply:
Thank you for your review. Memory leak is fixed.